sdk/mediaservices/Microsoft.Azure.Media.LiveVideoAnalytics.Edge/src/MediaGraphPemCertificateList.cs

```diff
@@ -0,0 +1,95 @@
+﻿// -----------------------------------------------------------------------
+//  <copyright company="Microsoft Corporation">
+//      Copyright (C) Microsoft Corporation. All rights reserved.
+//  </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+
+namespace Microsoft.Azure.Media.LiveVideoAnalytics.Edge.Models
+{
+    /// <summary>
+    /// Initializes a new instance of the MediaGraphPemCertificateList class.
+    /// </summary>
+    public partial class MediaGraphPemCertificateList
+    {
+        /// <summary>
+        /// How long the base 64 lines should be.
+        /// </summary>
+        private const int Base64LineLength = 76;
+
+        /// <summary>
+        /// Header for a PEM formatted certificate.
+        /// </summary>
+        /// <remarks>
+        /// See https://tools.ietf.org/html/rfc7468 Section 5.1.
+        /// </remarks>
+        private const string PemCertificateHeader = "-----BEGIN CERTIFICATE-----";
+
+        /// <summary>
+        /// Footer for a PEM formatted certificate.
+        /// </summary>
+        /// <remarks>
+        /// See https://tools.ietf.org/html/rfc7468 Section 5.1.
+        /// </remarks>
+        private const string PemCertificateFooter = "-----END CERTIFICATE-----";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MediaGraphPemCertificateList"/> class.
+        /// </summary>
+        /// <param name="certificates"> The list of certificates.</param>
+        public MediaGraphPemCertificateList(IList<X509Certificate2> certificates)
+        {
+            Certificates = certificates.Select(ConvertToPemString).ToList();
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MediaGraphPemCertificateList"/> class.
+        /// </summary>
+        /// <param name="certificates"> The certificates params.</param>
+        public MediaGraphPemCertificateList(params X509Certificate2[] certificates)
+        {
+            Certificates = certificates.Select(ConvertToPemString).ToList();
+        }
```

Comment:
Do you need both ctors?

Reply:
This was just a way of providing customers more options, for example if the customer wants to provide one certificate, they don't have to create a list.

Reply:
Does it make sense to have the `IList<>` one then?

Reply:
Yes, because in the other case where they might have a list of certificates, we give them the option to construct the object as well with a list.

Reply:
It seems like the second method (the one that takes a `params` ) could work on both. If the user has a list they can call that method with `myList.ToArray()`. Granted, there are some performance considerations, but assuming that API is not called a lot of times, it would be fine.

In either case, I am fine with this -- with the message that adding APIs is easy and removing them is impossible 🙂.